### DISCUSSION TITLE
What is an "inactive" query?

### DISCUSSION THREAD

**Discussion by vineetdigit** [TanStack/query#1913] [Q&A] [answered]
From the documentation below, it's not clear to me what an "**active instances of useQuery, ...**" means.

> Query results that have no more active instances of useQuery, useInfiniteQuery or query observers are labeled as "inactive" and remain in the cache in case they are used again at a later time.
> By default, "inactive" queries are garbage collected after 5 minutes.

is there an explanation (hopefully with some example) that explains this further?

This is important for me to understand to decide what flags to pass below

```
await queryClient.invalidateQueries('posts', {
   exact,
   refetchActive: true,
   refetchInactive: false
 }, { throwOnError })
```

**Comment by TkDodo** [comment, marked as answer]
I think this is a good definition by example: https://react-query.tanstack.com/guides/caching

Basically, every time you call `useQuery`, you create an `observer`. As long as your component that calls `useQuery` is mounted, the query is considered `active` (can be stale or fresh or fetching, but it's active).
When you unmount all observers (because of conditional rendering, or you navigate to a different page via client-side routing), you don't really need the data anymore - your query is now `inactive`. The data will still stay in the cache for 5 more minutes - or whatever you set as `cacheTime`. This is nice because when you come back to your page, making your query active again, you can again leverage the cache. After some time, the data is deleted for garbage-collection. Hope that explains it :)

**Reply by vineetdigit** [reply]
This has wonderfully explained what an inactive query means! Yes, it definitely helps. Thank you @TkDodo

**Reply by shivammalhotra859** [reply]
thankyou....

as long as component is mount query remains active ......but when component is unmounted query is marked as Inactive
and Inactive query is garbage collected after the "cacheTime" (default value of  cacheTime is 5 minutes )

**Reply by MarwanBz** [reply]
coming from the end of 2025, 
thanks @TkDodo for the amazing answer
